cmp/tile/TilingContainerModel.js

```diff
@@ -0,0 +1,219 @@
+    dimensions;
+
+    //-----------------------
+    // Immutable Properties
+    //-----------------------
+    content;
+    emptyText;
+    desiredRatio;
+    spacing;
+    minRatio;
+    maxRatio;
+    minWidth;
+    maxWidth;
+    minHeight;
+    maxHeight;
+
+    @computed.struct
+    get layout() {
+        return this.selectLayout();
+    }
+
+    /**
+     * @param {Object} c - TilingContainerModel configuration.
+     * @param {(ReactElement|Object|function)} c.content - Hoist Component (class or functional) to be rendered by this
+     *      for each tile; or function returning react element to be rendered for each tile. Will receive `data` as a prop.
+     * @param {string} [c.emptyText] - text to display when the container is empty.
+     * @param {number} [c.desiredRatio] - Desired tile width / height ratio (i.e. desiredRatio: 2 == twice as wide as tall).
+     *      Layouts will strive to meet this ratio, but the final ratio may vary. Defaults to 1 (i.e. square tiles).
+     * @param {number} [c.spacing] - The space between tiles (in px).
+     * @param {number} [c.minRatio] - Min tile ratio. Only layouts that meet this condition will be considered.
+     * @param {number} [c.maxRatio] - Max tile ratio. Only layouts that meet this condition will be considered.
+     * @param {number} [c.minWidth] - Min tile width (in px). Only layouts that meet this condition will be considered.
+     * @param {number} [c.maxWidth] - Max tile width (in px). Only layouts that meet this condition will be considered.
+     * @param {number} [c.minHeight] - Min tile height (in px). Unlike minWidth, failing to meet this condition does not
+     *      automatically invalidate the layout. Instead, the container will scroll if necessary.
+     * @param {number} [c.maxHeight] - Max tile height (in px). Unlike maxWidth, failing to meet this condition does not
+     *      automatically invalidate the layout. Instead, the container will scroll if necessary.
+     */
+    constructor({
+        content,
+        emptyText= 'No items found...',
+        desiredRatio = 1,
+        spacing = 0,
+        minRatio,
+        maxRatio,
+        minWidth,
+        maxWidth,
+        minHeight,
+        maxHeight
+    } = {}) {
+        super();
+        makeObservable(this);
+
+        this.content = content;
+        this.emptyText = emptyText;
+        this.desiredRatio = desiredRatio;
+        this.spacing = spacing;
+        this.minRatio = minRatio;
+        this.maxRatio = maxRatio;
+        this.minWidth = minWidth;
+        this.maxWidth = maxWidth;
+        this.minHeight = minHeight;
+        this.maxHeight = maxHeight;
+    }
+
+    /**
+     * Load a new and complete dataset.
+     *
+     * Data is an array of objects, models, etc. For each entry, a tile will be rendered using
+     * `content` and sized and placed according to the layout.
+     *
+     * @param {*[]} data - source data to load
+     */
+    @action
+    loadData(data) {
```

Comment:
wonder if this is really just a simple setter, if it might be more straightforward to call it setData?   

On the point about a store again, you could imagine it would be real nice to support setting a store here, and having an internal computed actually dereference (and observe) the actual records.